app/Controller/ServersController.php

```diff
@@ -2205,6 +2205,45 @@ public function dbSchemaDiagnostic()
         if (!$this->_isSiteAdmin()) {
             throw new MethodNotAllowedException(__('Only site admin accounts get the DB schema diagnostic.'));
         }
-        return $this->RestResponse->viewData($this->Server->dbSchemaDiagnostic(), $this->response->type());
+        $dbSchemaDiagnostics = $this->Server->dbSchemaDiagnostic();
+        if ($this->_isRest()) {
+            return $this->RestResponse->viewData($dbSchemaDiagnostics, $this->response->type());
+        } else {
+            $this->set('checkedTableColumn', $dbSchemaDiagnostics['checked_table_column']);
+            $this->set('dbSchemaDiagnostics', $dbSchemaDiagnostics['diagnostic']);
+            $this->set('expectedDbVersion', $dbSchemaDiagnostics['expected_db_version']);
+            $this->set('actualDbVersion', $dbSchemaDiagnostics['actual_db_version']);
+            $this->set('error', $dbSchemaDiagnostics['error']);
+            $this->set('remainingLockTime', $dbSchemaDiagnostics['remaining_lock_time']);
+            $this->set('updateFailNumberReached', $dbSchemaDiagnostics['update_fail_number_reached']);
+            $this->set('updateLocked', $dbSchemaDiagnostics['update_locked']);
+            $this->render('/Elements/healthElements/db_schema_diagnostic');
+        }
+
+    }
+
+    public function execSQLQuery() {
```

Comment:
Instead of executing the recovery scripts, I'd much rather just display it to the user and let them run it outside of MISP if they want. This is generally something that can be SLOW and them hammering away at their DB might have adverse effects. It shouldn't be something you trigger on a whim in some cases.

Reply:
I totally get your point. It's there to improve user experience.
I do not expected all site admins to be comfortable to interact with the database directly. The idea is to facilitate his task by providing a button to press.
Moreover, it can also be used by experienced admin that just want to quickly fix their DB issues.
In the end, the result will be the same. An SQL query will be executed.
If we keep this quick-fix-by-pressing-a-button feature, we definitely need to add a proper notice warning the user of the consequence.

Reply:
Just in case, warning message added

Reply:
Keep in mind a click on this when it comes to an alter table on a large instance's log table might completely kill it until mysql is restarted. I'd reaaaaally not want this directly accessible in the UI.

Reply:
Providing a copy-paste script and letting admins sql-dump, shut their misp off and run them at their own risk is fine. But a warning is definitely not enough to prevent screw ups with this.

Reply:
Alright. Removing that part of the code ;)